### PR TITLE
feat: Allow container linking for AWS ECS applications.

### LINKED ISSUE #2658: Update how ECS  docker id are handled in the agent
### Describe the story, please be clear on scope of the story.
See https://source.datanerd.us/agents/agent-specs/pull/686

If the ECS env vars are used to get the docker id, use a new field to send up the value, otherwise use the old location.  The format of the value can be anything, but on the backend they are looking for the standard id and the ECS formatted id.  This will require some new logic to handle which area to place the capture ids and new unit tests.

```
"vendors": {
        "ecs": {
          "ecsDockerId": "1e1698469422439ea356071e581e8545-2769485393"
        }
      },
```

### Acceptance Criteria
- Changes are complete with tests
- Manual testing of changes in ECS
- Optional/Hopeful: End-to-end testing has been done.

### Dependencies
- [ ] e2e testing depends on APM Services completing their work.

### Estimates
S (M with e2e testing)

### PR DESCRIPTION
**Pull request by jaffinito**
## Description

- Updates our utilization payload to match new version of the spec.
- Changes the logic for where we put the docker id captured from AWS ECS
- Expands unit tests for ECS

Closes #2658

# Author Checklist
- [ ] Unit tests, Integration tests, and Unbounded tests completed
- [ ] Performance testing completed with satisfactory results (if required)

# Reviewer Checklist
- [ ] Perform code review
- [ ] Pull request was adequately tested (new/existing tests, performance tests)

### CODE REVIEW

**Comment by jaffinito** [comment]
Leaving this in draft until APM Services can take a look at the data and confirm this is working for them.

**Comment by jaffinito** [comment]
Testing has been completed and the changes are working as expected.  I have set the spec change PR to ready for review and will be marking this one as ready once it gets some approvals.

File: src/Agent/NewRelic/Agent/Core/Utilization/EcsVendorModel.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+namespace NewRelic.Agent.Core.Utilization
+{
+    public class EcsVendorModel : IVendorModel
+    {
+        private readonly string _ecsDockerId;
+
+        public string VendorName { get { return "ecs"; } }
+
+        [JsonProperty("ecsDockerId", NullValueHandling = NullValueHandling.Ignore)]
```
Comment (jaffinito) [inline src/Agent/NewRelic/Agent/Core/Utilization/EcsVendorModel.cs:14]: If this is null, the whole vendor is ignored since it is the only field.

**Review by tippmar-nr — DISMISSED** [review]
Looks good!

**Review by chynesNR — DISMISSED** [review]

**Review by nrcventura — DISMISSED** [review]

**Review by nrcventura — APPROVED** [review]

**Review by chynesNR — APPROVED** [review]

**Pull request merged**